### ISSUE TITLE
Unmount when lid closes

### ISSUE THREAD

**Issue by nielsmouthaan** [nielsmouthaan/ejectify-macos#16]
Feature request: extend "Unmount when" options with the ability to unmount when the lid closes.

**Comment by shepazon** [comment]
Yes please! It's easily the most common way my Mac goes to sleep. Though I suppose I could change that habit, I kind of don't want to. :)

**Comment by aheiay** [comment]
Kind of confused. I still got the "Disk unmount properly" error message when I close / open the lid of my MacBook pro. 

If closing the Lid is equal to SLEEP?

**Comment by nielsmouthaan** [comment]
Not perse. It depends on how your Mac is configured. By default, it should be.

Have you tried other "Unmount when" options?

**Comment by aheiay** [comment]
I am sure the configuration in power management is set to sleep when close the lid, but the error still showing up.

Using OS 10.15.7 here

**Comment by nielsmouthaan** [comment]
How have you connected your drive? E.g. via a hub, or directly in the Mac's port?

**Comment by wdool** [comment]
@nielsmouthaan which option do I have to select, or is this not included in the last update?

**Comment by nielsmouthaan** [comment]
> @nielsmouthaan which option do I have to select, or is this not included in the last update?

The ability to unmount when the lid closes? This is not (yet) implemented.

**Comment by robled** [comment]
I bought the app hoping to just close the lid and then unplug it from my dock that has an external monitor and my drives connected.    

However, I've found that if you have an external monitor connected that the Mac goes into "clamshell" mode and therefore you can't trigger any of the events the app supports.  Would it be possible to add lid close event support?

**Comment by nielsmouthaan** [comment]
Reopened. Wil look into this for a future update.
